deeplinkdispatch-processor/src/test/java/com/airbnb/deeplinkdispatch/DocumentorTest.java

```diff
@@ -0,0 +1,100 @@
+    @Mock private ProcessingEnvironment processingEnv;
+    @Mock private Messager messager;
+    @Mock private Map<String, String> options;
+    @Mock private Elements elements;
+
+    @Before public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        when(processingEnv.getMessager()).thenReturn(messager);
+        when(processingEnv.getOptions()).thenReturn(options);
+        when(processingEnv.getElementUtils()).thenReturn(elements);
+    }
+
+    @Test public void testInitWithRightFilePath() {
+        when(options.get(Documentor.DOC_OUTPUT_PROPERTY_NAME)).thenReturn(FILE_PATH);
+        Documentor documentor = new Documentor(processingEnv);
+        assertThat(documentor.getFile()).isNotNull();
+    }
+
+    @Test public void testInitWithoutFilePath() {
+        Documentor documentor = new Documentor(processingEnv);
+        assertThat(documentor.getFile()).isNull();
+    }
+
+    @Test public void testWrite() {
+        when(options.get(Documentor.DOC_OUTPUT_PROPERTY_NAME)).thenReturn(FILE_PATH);
+        Documentor documentor = new Documentor(processingEnv);
+        documentor.write(getElements());
+        try {
+            String actual = Files.toString(new File(FILE_PATH), Charsets.UTF_8);
+            String expected = "airbnb://example.com/{foo}/bar\\n|#|\\nSample doc\\n|#|\\nDocClass\\n|##|\\n" +
+                    "airbnb://example.com/{foo}/bar\\n|#|\\n\\n|#|\\nDocClass#DocMethod\\n|##|\\n";
+            assertEquals(expected, actual);
+        } catch (IOException e) {
+            fail();
+        }
+    }
+
+    private List<DeepLinkAnnotatedElement> getElements() {
+        TypeElement element1 = mock(TypeElement.class);
+        Name name1 = mock(Name.class);
+        when(elements.getDocComment(element1)).thenReturn("Sample doc \n @param empty \n @return nothing");
+        when(element1.getSimpleName()).thenReturn(name1);
+        when(name1.toString()).thenReturn("DocClass");
+
+        TypeElement element2 = mock(TypeElement.class);
+        TypeElement element2Enclosed = mock(TypeElement.class);
+        Name name2 = mock(Name.class);
+        Name name2Enclosed = mock(Name.class);
+        when(element2.getSimpleName()).thenReturn(name2);
+        when(name2.toString()).thenReturn("DocMethod");
+        when(element2.getEnclosingElement()).thenReturn(element2Enclosed);
+        when(element2Enclosed.getSimpleName()).thenReturn(name2Enclosed);
+        when(name2Enclosed.toString()).thenReturn("DocClass");
+
+        try {
+            DeepLinkAnnotatedElement deepLinkElement1 =
+                new DeepLinkAnnotatedElement("airbnb://example.com/{foo}/bar", element1, DeepLinkEntry.Type.CLASS);
+            DeepLinkAnnotatedElement deepLinkElement2 =
+                    new DeepLinkAnnotatedElement("airbnb://example.com/{foo}/bar", element2, DeepLinkEntry.Type.METHOD);
+
+            return ImmutableList.of(deepLinkElement1, deepLinkElement2);
+        } catch (MalformedURLException e) {
```

Comment:
just add `MalformedURLException` to the method `throws` so you don't need this empty catch block